From 1205f7fc1e1e38cb2fc56560c641af9395ab0ce5 Mon Sep 17 00:00:00 2001
From: Stephen Boyd <sboyd@codeaurora.org>
Date: Thu, 25 Apr 2013 10:16:28 -0700
Subject: [PATCH] msm: thermal: Initialize sysfs attributes to quiet lockdep

Dynamically allocated sysfs attributes need to be initialized
with sysfs_attr_init() otherwise we get

 BUG: key ee1d8748 not in .data!

messages printed and lockdep becomes useless.

Change-Id: I74d9b6556f50d7e578569f67bb2c6b9b874cd64f
Signed-off-by: Stephen Boyd <sboyd@codeaurora.org>
---
 drivers/thermal/msm_thermal.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/thermal/msm_thermal.c b/drivers/thermal/msm_thermal.c
index 12ac3bc..42119aa 100644
--- a/drivers/thermal/msm_thermal.c
+++ b/drivers/thermal/msm_thermal.c
@@ -103,6 +103,7 @@ enum PMIC_SW_MODE {
 	ko_attr.attr.mode = 444; \
 	ko_attr.show = vdd_rstr_reg_##_name##_show; \
 	ko_attr.store = NULL; \
+	sysfs_attr_init(&ko_attr.attr); \
 	_rail.attr_gp.attrs[j] = &ko_attr.attr;
 
 #define VDD_RES_RW_ATTRIB(_rail, ko_attr, j, _name) \
@@ -110,6 +111,7 @@ enum PMIC_SW_MODE {
 	ko_attr.attr.mode = 644; \
 	ko_attr.show = vdd_rstr_reg_##_name##_show; \
 	ko_attr.store = vdd_rstr_reg_##_name##_store; \
+	sysfs_attr_init(&ko_attr.attr); \
 	_rail.attr_gp.attrs[j] = &ko_attr.attr;
 
 #define VDD_RSTR_ENABLE_FROM_ATTRIBS(attr) \
@@ -126,6 +128,7 @@ enum PMIC_SW_MODE {
 	ko_attr.attr.mode = 644; \
 	ko_attr.show = psm_reg_##_name##_show; \
 	ko_attr.store = psm_reg_##_name##_store; \
+	sysfs_attr_init(&ko_attr.attr); \
 	_rail.attr_gp.attrs[j] = &ko_attr.attr;
 
 #define PSM_REG_MODE_FROM_ATTRIBS(attr) \
